### ISSUE TITLE
Load smap file from filename regex

### ISSUE THREAD

**Issue by rafaqz** [rafaqz/GeoData.jl#46]
Getting the date from the HDF5 is too slow.

**Comment by jamesmaino** [comment]
Can we increase the priority of this one? Adds a good 30 mins to the start up every time I use the full SMAP data set.

**Comment by rafaqz** [comment]
Wow 30 mins.

You ou can do it by just regex filtering the filenames

```julia
smapfiles = joinpath.(smap_folder, [f for f in readdir(smapfolder) if occursin(r".h5$", f) && occursin("_2017" , f)])
smapseries = SMAPseries(smapfiles)
```

See

https://github.com/cesaraustralia/SpottedWingDrosophila/blob/master/establishment.jmd#L270

**Comment by rafaqz** [comment]
SMAPseries actually takes a `Vector` of `String`:

https://github.com/rafaqz/GeoData.jl/blob/master/src/sources/smap.jl#L114

**Comment by jamesmaino** [comment]
Oh I didn't realise it could already taken a vector of strings. I think this is enough for now but would be useful in future, e.g. for AWAP grids or something were there are lots of grid files.

**Comment by rafaqz** [comment]
closed by #54 so this is automatic
